<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com)
     Commit Version: 1.5.1-0-g916b5db  Build Version: 1.5.7152.31018
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="eight_hollow_cubes">
  <link
    name="cube1">
    <inertial>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <mass
        value="0.1" />
      <inertia
        ixx="6.023E-05"
        ixy="0"
        ixz="0"
        iyy="7.88E-05"
        iyz="0"
        izz="6.023E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube1.STL" />
      </geometry>
    </collision>
  </link>

  <link
    name="cube2">
    <inertial>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <mass
        value="0.1" />
      <inertia
        ixx="6.023E-05"
        ixy="0"
        ixz="0"
        iyy="7.88E-05"
        iyz="0"
        izz="6.023E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube2.STL" />
      </geometry>
    </collision>
  </link>

  <joint
    name="joint1"
    type="revolute">
    <origin
      xyz="0.025 0.025 0.025"
      rpy="1.5708 0 0" />
    <parent
      link="cube1" />
    <child
      link="cube2" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.1416"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>

  <link
    name="cube3">
    <inertial>
      <origin
        xyz="-0.025 -0.025 -0.025"
        rpy="0 0 0" />
      <mass
        value="0.1" />
      <inertia
        ixx="6.023E-05"
        ixy="0"
        ixz="0"
        iyy="7.88E-05"
        iyz="0"
        izz="6.023E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube3.STL" />
      </geometry>
    </collision>
  </link>

  <joint
    name="joint2"
    type="revolute">
    <origin
      xyz="0.05 0 0"
      rpy="0 0 0" />
    <parent
      link="cube2" />
    <child
      link="cube3" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="3.1416"
      effort="0"
      velocity="0" />
  </joint>

  <link
    name="cube4">
    <inertial>
      <origin
        xyz="-0.025 -0.025 -0.025"
        rpy="0 0 0" />
      <mass
        value="0.1" />
      <inertia
        ixx="6.023E-05"
        ixy="0"
        ixz="0"
        iyy="7.88E-05"
        iyz="0"
        izz="6.023E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube4.STL" />
      </geometry>
    </collision>
  </link>

  <joint
    name="joint3"
    type="revolute">
    <origin
      xyz="0 0 -0.0500000000000001"
      rpy="0 0 0" />
    <parent
      link="cube3" />
    <child
      link="cube4" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="0"
      upper="3.1416"
      effort="0"
      velocity="0" />
  </joint>

  <link
    name="cube5">
    <inertial>
      <origin
        xyz="-0.025 -0.025 -0.025"
        rpy="0 0 0" />
      <mass
        value="0.1" />
      <inertia
        ixx="6.023E-05"
        ixy="0"
        ixz="0"
        iyy="7.88E-05"
        iyz="0"
        izz="6.023E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube5.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube5.STL" />
      </geometry>
    </collision>
  </link>

  <joint
    name="joint4"
    type="revolute">
    <origin
      xyz="0 0 -0.05"
      rpy="0 0 0" />
    <parent
      link="cube4" />
    <child
      link="cube5" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="3.1416"
      effort="0"
      velocity="0" />
  </joint>

  <link
    name="cube6">
    <inertial>
      <origin
        xyz="-0.025 -0.025 -0.025"
        rpy="0 0 0" />
      <mass
        value="0.1" />
      <inertia
        ixx="6.023E-05"
        ixy="0"
        ixz="0"
        iyy="7.88E-05"
        iyz="0"
        izz="6.023E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube6.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube6.STL" />
      </geometry>
    </collision>
  </link>

  <joint
    name="joint5"
    type="revolute">
    <origin
      xyz="-0.05 0 0"
      rpy="0 0 0" />
    <parent
      link="cube5" />
    <child
      link="cube6" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.1416"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>

  <link
    name="cube7">
    <inertial>
      <origin
        xyz="0.025 -0.025 0.025"
        rpy="0 0 0" />
      <mass
        value="0.1" />
      <inertia
        ixx="6.023E-05"
        ixy="0"
        ixz="0"
        iyy="7.88E-05"
        iyz="0"
        izz="6.023E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube7.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube7.STL" />
      </geometry>
    </collision>
  </link>

  <joint
    name="joint6"
    type="revolute">
    <origin
      xyz="-0.05 0 0"
      rpy="0 0 0" />
    <parent
      link="cube6" />
    <child
      link="cube7" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-3.1416"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>

  <link
    name="cube8">
    <inertial>
      <origin
        xyz="0.025 -0.025 0.025"
        rpy="0 0 0" />
      <mass
        value="0.1" />
      <inertia
        ixx="6.023E-05"
        ixy="0"
        ixz="0"
        iyy="7.88E-05"
        iyz="0"
        izz="6.023E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube8.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://eight_hollow_cubes/meshes/cube8.STL" />
      </geometry>
    </collision>
  </link>

  <joint
    name="joint7"
    type="revolute">
    <origin
      xyz="0 0 0.05"
      rpy="0 0 0" />
    <parent
      link="cube7" />
    <child
      link="cube8" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="0"
      upper="3.1416"
      effort="0"
      velocity="0" />
  </joint>
</robot>
